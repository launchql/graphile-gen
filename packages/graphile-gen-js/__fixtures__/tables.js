export default [
  {
    name: 'test1',
    defn: {
      databaseId: '5b3263d5-13eb-491c-b397-7f227edd3cf2',
      name: 'products',
      id: 'a65d1d03-e302-41a9-11c9-18470c9d7f46',
      fields: {
        nodes: [
          {
            name: 'id',
            defaultValue: 'uuid_generate_v4()',
            isHidden: false,
            isRequired: true,
            type: 'uuid'
          },
          {
            name: 'owner_id',
            isHidden: false,
            defaultValue: 'get_current_user_id()',
            isRequired: true,
            type: 'uuid'
          },
          {
            name: 'name',
            isHidden: false,
            isRequired: true,
            type: 'text'
          },
          {
            name: 'rhino_foot',
            description: null,
            isHidden: false,
            isRequired: false,
            type: 'text'
          },
          {
            name: 'hidden_foot',
            description: null,
            isHidden: true,
            isRequired: false,
            type: 'text'
          },
          {
            name: 'lizard_feet',
            isHidden: false,
            isRequired: false,
            type: 'text'
          }
        ]
      }
    }
  }
];
