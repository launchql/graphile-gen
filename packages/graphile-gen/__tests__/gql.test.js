import {
  updateOne,
  createOne,
  deleteOne,
  getOne,
  getMany,
  getManyOwned
} from '../src';
import cases from 'jest-in-case';
import { print } from 'graphql';
import tables from '../__fixtures__/tables';

cases(
  'getOne',
  options => {
    const value = getOne(options.defn);
    expect(print(value)).toMatchSnapshot();
  },
  tables
);

cases(
  'getMany',
  options => {
    const value = getMany(options.defn);
    expect(print(value)).toMatchSnapshot();
  },
  tables
);

cases(
  'getManyOwned',
  options => {
    const value = getManyOwned(options.defn, 'owner_id');
    expect(print(value)).toMatchSnapshot();
  },
  tables
);

cases(
  'updateOne',
  options => {
    const value = updateOne(options.defn);
    expect(print(value)).toMatchSnapshot();
  },
  tables
);

cases(
  'createOne',
  options => {
    const value = createOne(options.defn);
    expect(print(value)).toMatchSnapshot();
  },
  tables
);

cases(
  'deleteOne',
  options => {
    const value = deleteOne(options.defn);
    expect(print(value)).toMatchSnapshot();
  },
  tables
);
