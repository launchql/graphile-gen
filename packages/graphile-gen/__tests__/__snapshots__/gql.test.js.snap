// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createOne test1 1`] = `
"mutation createProductMutation($id: UUID, $ownerId: UUID, $name: String!, $rhinoFoot: String, $hiddenFoot: String, $lizardFeet: String) {
  createProduct(input: {product: {id: $id, ownerId: $ownerId, name: $name, rhinoFoot: $rhinoFoot, hiddenFoot: $hiddenFoot, lizardFeet: $lizardFeet}}) {
    product {
      id
      ownerId
      name
      rhinoFoot
      hiddenFoot
      lizardFeet
    }
  }
}
"
`;

exports[`deleteOne test1 1`] = `
"mutation deleteProductMutation($id: UUID!) {
  deleteProduct(input: {id: $id}) {
    product {
      id
      ownerId
      name
      rhinoFoot
      hiddenFoot
      lizardFeet
    }
  }
}
"
`;

exports[`getMany test1 1`] = `
"query getProductsQuery {
  products {totalCount, nodes: {
    id
    ownerId
    name
    rhinoFoot
    hiddenFoot
    lizardFeet
  }}
}
"
`;

exports[`getManyOwned test1 1`] = `
"query getProductsOwnedByOwnerIdQuery($ownerId: UUID!) {
  products(condition: {ownerId: $ownerId}) {totalCount, nodes: {
    id
    ownerId
    name
    rhinoFoot
    hiddenFoot
    lizardFeet
  }}
}
"
`;

exports[`getOne test1 1`] = `
"query getProductQuery($id: UUID!) {
  product(id: $id) {
    id
    ownerId
    name
    rhinoFoot
    hiddenFoot
    lizardFeet
  }
}
"
`;

exports[`updateOne test1 1`] = `
"mutation updateProductMutation($id: UUID!, $ownerId: UUID, $name: String, $rhinoFoot: String, $hiddenFoot: String, $lizardFeet: String) {
  updateProduct(input: {id: $id, patch: {ownerId: $ownerId, name: $name, rhinoFoot: $rhinoFoot, hiddenFoot: $hiddenFoot, lizardFeet: $lizardFeet}}) {
    product {
      id
      ownerId
      name
      rhinoFoot
      hiddenFoot
      lizardFeet
    }
  }
}
"
`;
